src/mbed_tools/project/_internal/libraries.py

```diff
@@ -42,6 +42,10 @@ def get_git_reference(self) -> git_utils.GitReference:
         """
         raw_ref = self.reference_file.read_text().strip()
         url, sep, ref = raw_ref.partition("#")
+
+        if url.endswith("/"):
```

Comment:
Could we add a comment explaining why this is needed for future reference?

Reply:
I think the commit explains it, should I link the github issue?

Reply:
Ah, you linked to the issue in the commit message. Resolving.